src/main/java/io/pakland/mdas/githubstats/infrastructure/rest/repository/WebClientConfiguration.java

```diff
@@ -0,0 +1,25 @@
+package io.pakland.mdas.githubstats.infrastructure.rest.repository;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+
+@Component
+public class WebClientConfiguration {
+
+    private final WebClient webClient;
+
+    public WebClientConfiguration(String baseUrl, String apiKey) {
+        this.webClient = WebClient.builder().baseUrl(baseUrl)
+                .defaultHeaders(httpHeaders -> {
+                    httpHeaders.set(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE);
+                    httpHeaders.set(HttpHeaders.AUTHORIZATION, "Bearer " + apiKey);
```

Comment:
Is this GitHub specific? If it is the case, I think we should probably rename it to `GitHubWebClientConfiguration` or something similar. Thoughts?